swift/Samples/TicTacToe/Sources/Game/RunGameWorkflow.swift

```diff
@@ -115,11 +124,40 @@ extension RunGameWorkflow {
                     leftItem: BackStackScreen.BarContent.BarButtonItem.button(BackStackScreen.BarContent.Button(
                         content: .text("Quit"),
                         handler: {
-                            sink.send(.back)
+                            sink.send(.confirmQuit)
+                        }))))))
+            
+        case .maybeQuit:
+            
+            let takeTurnsScreen = TakeTurnsWorkflow(
+                playerX: state.playerX,
+                playerO: state.playerO)
+                .rendered(with: context)
+            backStackItems.append(BackStackScreen.Item(
+                screen: takeTurnsScreen,
+                barVisibility: .visible(BackStackScreen.BarContent(
+                    leftItem: BackStackScreen.BarContent.BarButtonItem.button(BackStackScreen.BarContent.Button(
+                        content: .text("Quit"),
+                        handler: {
+                            sink.send(.confirmQuit)
                         }))))))
+            
+            let confirmQuitScreen = ConfirmQuitWorkflow()
+                .mapOutput( { output -> Action in
+                    switch output {
+                    case .cancel:
+                        return .startGame
+                    case .confirm:
+                        return .back
+                    }
+                })
+                .rendered(with: context)
+            modals.append(ModalContainerScreen.Modal(screen: AnyScreen(confirmQuitScreen), style: .sheet, key: "0", animated: true))
         }
+        
+        let modalContainerScreen = ModalContainerScreen(baseScreen: AnyScreen(BackStackScreen(items: backStackItems)), modals: modals)
```

Comment:
Do we need the `AnyScreen(BackStackScreen(items: backStackItems))`, I would think the base doesn't need to be wrapped in an AnyScreen.

Reply:
Since i specified the return type as `ModalContainerScreen<AnyScreen>` where type of BaseScreen is specified as AnyScreen, i have to wrap `BackStackScreen(items: backStackItems)` into `AnyScreen`.  Please let me know if i am missing something here since i am new to Swift.

Reply:
Ah, is there any benefit to have the return type of ModalContainerScreen<AnyScreen> vs ModalContainerScreen<BackStackScreen>. cc: @bencochran, @dhavalshreyas

Reply:
+1 to minimizing `AnyScreen` usage wherever possible.

Reply:
ok, makes sense.